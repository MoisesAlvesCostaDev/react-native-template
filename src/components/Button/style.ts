import styled from 'styled-components/native';

import {BUTTON_BACKGROUND_COLOR,BUTTON_TEXT_COLOR} from "assets/colors";

export const StyledTouchableOpacity = styled.TouchableOpacity`
  background-color:${BUTTON_BACKGROUND_COLOR};
  height: 48px;
  align-items: center;
  justify-content:center;
  border-radius:12px;
`;

export const StyledText = styled.Text`
  color:${BUTTON_TEXT_COLOR};
  font-size:16px;
`;


