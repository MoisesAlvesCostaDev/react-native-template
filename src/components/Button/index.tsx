import React from "react";
import { StyledTouchableOpacity, StyledText } from "./style";

export default function Button({ title = "Button title", ...props }) {
  return (
    <StyledTouchableOpacity testID="buttonStyled" {...props}>
      <StyledText>{title}</StyledText>
    </StyledTouchableOpacity>
  );
}
