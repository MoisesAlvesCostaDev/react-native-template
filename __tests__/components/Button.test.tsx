import "react-native";
import React from "react";
import Button from "components/Button";
import { fireEvent, render } from "@testing-library/react-native";

describe("test render button component with title button", () => {
  it("renders correctly", () => {
    const { getByTestId, getByText, toJSON } = render(
      <Button title="button" />
    );
    getByText("button");
    expect(toJSON()).toMatchSnapshot();
  });
});

describe("test render and click button component", () => {
  it("renders correctly", () => {
    const onClick = jest.fn();
    const { getByTestId } = render(<Button title="button" onPress={onClick} />);
    const buttonStyled = getByTestId("buttonStyled");
    console.log(buttonStyled);
    fireEvent.press(buttonStyled);
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
