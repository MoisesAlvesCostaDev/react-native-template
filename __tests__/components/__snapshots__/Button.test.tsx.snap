// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render button component with title button renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#EDEDED",
      "borderRadius": 12,
      "height": 48,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
  testID="buttonStyled"
>
  <Text
    style={
      Array [
        Object {
          "color": "#646464",
          "fontSize": 16,
        },
      ]
    }
  >
    button
  </Text>
</View>
`;
